test/unit/test_annotated_code.c

```diff
@@ -0,0 +1,305 @@
+	r_vector_push (test_annotations, &annotation); //8
+	annotation = make_code_annotation (58, 64, R_CODE_ANNOTATION_TYPE_OFFSET, 4447, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (test_annotations, &annotation); //9
+
+	return test_annotations;
+}
+
+static RAnnotatedCode *get_hello_world() {
+	char *test_string = malloc (71 * sizeof (char));
+	strcpy (test_string, "\nvoid main(void)\n{\n    sym.imp.puts(\"Hello, World!\");\n    return;\n}\n");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+
+	RVector /*<RCodeAnnotation>*/ *test_annotations;
+	test_annotations = get_annotations_for_hello_world ();
+	RCodeAnnotation *annotation;
+	r_vector_foreach (test_annotations, annotation) {
+		r_annotated_code_add_annotation (code, annotation);
+	}
+	return code;
+}
+
+static bool test_r_annotated_code_new() {
+	//Testing RAnnoatedCode->code
+	char *test_string = malloc (14 * sizeof (char));
+	strcpy (test_string, "How are you?");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+	mu_assert_streq (code->code, test_string, "Code in RAnnotatedCode is not set as expected");
+
+	// Testing RAnnoatedCode->annotations
+	mu_assert_eq (code->annotations.elem_size, sizeof (RCodeAnnotation), "Code Annotations are initialized is not properly");
+
+	mu_end;
+}
+
+static bool test_r_annotated_code_free() {
+	char *test_string = malloc (14 * sizeof (char));
+	strcpy (test_string, "How are you?");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+
+	RCodeAnnotation test_annotation1, test_annotation2;
+	test_annotation1 = make_code_annotation (1, 2, R_CODE_ANNOTATION_TYPE_OFFSET, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&code->annotations, &test_annotation1);
+	test_annotation2 = make_code_annotation (1, 5, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&code->annotations, &test_annotation2);
+
+	// This test is only for run errors
+	r_annotated_code_free (code);
+
+	mu_end;
+}
+
+static bool test_equal(RCodeAnnotation *first, RCodeAnnotation *second) { // First - Got, Second - Expected
+	mu_assert_eq (first->start, second->start, "start of annotations doesn't match");
+	mu_assert_eq (first->end, second->end, "end of annotations doesn't match");
+	mu_assert_eq (first->type, second->type, "type of annotation doesn't match");
+	if (first->type == R_CODE_ANNOTATION_TYPE_OFFSET) {
+		mu_assert_eq (first->offset.offset, second->offset.offset, "offset of annotations doesn't match");
+	}
+	if (first->type == R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT) {
+		mu_assert_eq (first->syntax_highlight.type, second->syntax_highlight.type, "syntax highlight type of offset doesn't match");
+	}
+}
+
+static bool test_r_annotated_code_add_annotation() {
+	char *test_string = malloc (54 * sizeof (char));
+	strcpy (test_string, "abcdefghijklmnopqrtstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ");
```

Comment:
As reported by valgrind:
```
==28999== Invalid write of size 1
==28999==    at 0x10A568: test_r_annotated_code_add_annotation (test_annotated_code.c:143)
==28999==    by 0x10BA3F: all_tests (test_annotated_code.c:293)
==28999==    by 0x10BC1A: main (test_annotated_code.c:304)
==28999==  Address 0x625d716 is 0 bytes after a block of size 54 alloc'd
==28999==    at 0x483977F: malloc (vg_replace_malloc.c:307)
==28999==    by 0x10A4FF: test_r_annotated_code_add_annotation (test_annotated_code.c:142)
==28999==    by 0x10BA3F: all_tests (test_annotated_code.c:293)
==28999==    by 0x10BC1A: main (test_annotated_code.c:304)
==28999==
```
Instead, you can simply do:
```c
char *test_string = strdup ("abcdefghijklmnopqrtstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ");
```
The same goes for all the other malloc/strcpy cases.